### PR TITLE
netcdf 4.6.0 with opendap capabilities

### PR DESCRIPTION
**Pull request by pmjherman**
Until now, there was no MINGW build of netcdf with the capability of opendap access to remote thredds servers. The only possibility in Windows was to use a binary produced with Microsoft Visual C by the people issuing netcdf. This prevented, however, incorporation into R and maybe other MINGW-based software.

In this build, I have patched the netcdf software to allow for opendap, and changed the curl build slightly in order to make a build possible.

### CODE REVIEW

**Comment by pmjherman** [comment]
The build of continuous-integration/tea failed because it did not have a cmake version that was recent enough. It does work, however, with a recent MSYS2 system where cmake is up to date.
I have tested the build on a multitude of netcdf and opendap files and found no problem. I have also used the library successfully to build an R package.

File: mingw-w64-netcdf/PKGBUILD [hunk 1]
```diff
@@ -29,13 +46,17 @@ build() {
...
-    -DCMAKE_BUILD_TYPE=Release \
+	-DCMAKE_BUILD_TYPE=Release \
     -DBUILD_SHARED_LIBS=OFF \
     -DENABLE_DLL=OFF \
-    -DENABLE_DAP=OFF \
+    -DENABLE_DAP=ON \
     -DBUILD_TESTING=OFF \
     -DENABLE_DYNAMIC_LOADING=OFF \
     -DENABLE_NETCDF_4=ON \
+    -DZLIB_LIBRARY:FILEPATH=${MINGW_PREFIX}/lib/libz.a \
```
Comment (Alexpux) [inline mingw-w64-netcdf/PKGBUILD]: I will not accept forcing static linking. We use shared linking by default

File: mingw-w64-netcdf/PKGBUILD [hunk 2]
```diff
@@ -44,13 +65,18 @@ build() {
...
+	 -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
     -DCMAKE_BUILD_TYPE=Release \
     -DBUILD_SHARED_LIBS=ON \
-    -DENABLE_DAP=OFF \
+    -DENABLE_DAP=ON \
     -DBUILD_TESTING=OFF \
     -DENABLE_DYNAMIC_LOADING=OFF \
     -DENABLE_NETCDF_4=ON \
+	-DENABLE_LOGGING=ON \
+    -DZLIB_LIBRARY:FILEPATH=${MINGW_PREFIX}/lib/libz.a \
```
Comment (Alexpux) [inline mingw-w64-netcdf/PKGBUILD]: The same

File: mingw-w64-netcdf/PKGBUILD [hunk 3]
```diff
@@ -44,13 +65,18 @@ build() {
 
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
   ${MINGW_PREFIX}/bin/cmake -G "MSYS Makefiles" \
-    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
+	 -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
```
Comment (Alexpux) [inline mingw-w64-netcdf/PKGBUILD]: Dont use TAB's

File: mingw-w64-netcdf/PKGBUILD [hunk 4]
```diff
@@ -29,13 +46,17 @@ build() {
   MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
   ${MINGW_PREFIX}/bin/cmake -G "MSYS Makefiles" \
     -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
-    -DCMAKE_BUILD_TYPE=Release \
+	-DCMAKE_BUILD_TYPE=Release \
```
Comment (Alexpux) [inline mingw-w64-netcdf/PKGBUILD]: Dont use TABs

**Comment by Alexpux** [comment]
@pmjherman fix my comments

**Comment by pmjherman** [comment]
Thanks for the review and sorry for my beginner's mistakes. I have changed the editorial things (TABs). I found out that the compiler links to the libs could be left out alltogether. I added them with a previous version of netcdf, I think they may have been needed because of a bug in the cmake system at that time. Anyway, this also solves the problem of static linking. I hope that with this all will be OK.
I will commit my changes and then give a new pull request

**Comment by Alexpux** [comment]
Rebase changes on top of cerrent state. Also netcdf 4.6.1 releaseed

**Comment by Alexpux** [comment]
@pmjherman ping?

**Comment by Alexpux** [comment]
Fixed by https://github.com/Alexpux/MINGW-packages/commit/47e5d7d4c15510c3d092a6dfefd8a109bc03b78a

**Pull request closed without merging**